src/components/task/external.cjsx

```diff
@@ -0,0 +1,49 @@
+React = require 'react'
+
+
+{TaskStore, TaskActions} = require '../../flux/task'
+{TaskStepActions} = require '../../flux/task-step'
+
+LoadableItem = require '../loadable-item'
+
+# This component will only be mounted if the external link
+# was right-clicked on and opened in a new tab.
+#
+# Normal "left-click" usage of the link will be intercepted
+# and the external task opened immediately
+ExternalTask = React.createClass
+
+  getDefaultProps: ->
+    redirectToUrl: (url) -> window.location.href = url
```

Comment:
No, I don't think so.  For this component the redirect is called in `componentWillMount`, not in response to a click.  It's done that way because this component will only be mounted if it's opened in a new tab.

The reason the function is set as a prop-with-default is so the implementation can be swapped out in the spec to ensure that it's called.  Karma gets angry if `location.href` is changed mid-spec run.